src/s2wasm.h

```diff
@@ -159,16 +159,23 @@ class S2WasmBuilder {
     return cashew::IString(str.c_str(), false);
   }
 
-  int32_t getInt() {
-    int32_t ret = 0;
+  uint32_t getInt() {
```

Comment:
Another possible solution is to leave ret as int32_t but replace "if (neg) ret = -ret;" below with "if (neg) ret = -uint32_t(ret);".